src/konserve/memory.cljc

```diff
@@ -1,53 +1,91 @@
 (ns konserve.memory
   "Address globally aggregated immutable key-value store(s)."
-  (:require #?(:clj [clojure.core.async :refer [go <!!]])
-            [konserve.protocols :refer [PEDNAsyncKeyValueStore
-                                        -get -get-meta -update-in -dissoc -assoc-in
-                                        PBinaryAsyncKeyValueStore
-                                        -update-in
-                                        PBinaryAsyncKeyValueStore
-                                        -bget -bassoc
-                                        PKeyIterable
-                                        -keys]]
-            [clojure.core.async :as async :refer [go]]))
+  (:require [konserve.protocols :refer [PEDNAsyncKeyValueStore -update-in
+                                        PBinaryAsyncKeyValueStore PKeyIterable]]
+            #?(:clj [konserve.utils :refer [async+sync]])
+            [clojure.core.async :as async :refer [go <!]])
+  #?(:cljs (:require-macros [konserve.utils :refer [async+sync]])))
 
 (defrecord MemoryStore [state read-handlers write-handlers locks]
   PEDNAsyncKeyValueStore
-  (-exists? [this key] (go (if (@state key) true false)))
-  (-get [this key] (go (second (get @state key))))
-  (-get-meta [this key] (go (first (get @state key))))
-  (-update-in [this key-vec meta-up-fn up-fn args]
-    (go
-      (let [[fkey & rkey] key-vec
-            update-atom
-            (fn [store]
-              (swap! store
-                     (fn [old]
-                       (update old fkey
-                               (fn [[meta data]]
-                                 [(meta-up-fn meta)
-                                  (if rkey
-                                    (apply update-in data rkey up-fn args)
-                                    (apply up-fn data args))])))))]
-        [(second (get @state fkey))
-         (second (get (update-atom state) fkey))])))
-  (-assoc-in [this key-vec meta val] (-update-in this key-vec meta (fn [_] val) []))
-  (-dissoc [this key] (go (swap! state dissoc key) nil))
+  (-exists? [_ key opts]
+    (let [{:keys [sync?]} opts]
+      (async+sync sync?
+                  {go do
+                   <! do}
+                  (go  (if (get @state key false) true false)))))
+  (-get [_ key opts]
+    (let [{:keys [sync?]} opts]
+      (async+sync sync?
+                  {go do
+                   <! do}
+                  (go (second (get @state key))))))
+  (-get-meta [_ key opts]
+    (let [{:keys [sync?]} opts]
+      (async+sync sync?
+                  {go do
+                   <! do}
```

Comment:
Still open...

Reply:
Fixed. Note that the code in `memory.cljc` is also written for clarity of how the semantics work in terms of Clojure (and referenced in the documentation), not for maximum compactness (otherwise I would have used the general map here as well).